lib/collection/src/shards/replica_set.rs

```diff
@@ -818,9 +818,15 @@ impl ShardOperation for ShardReplicaSet {
 
         let (successes, failures): (Vec<_>, Vec<_>) = all_res.into_iter().partition_result();
 
-        // report all failing peers to consensus
-        for (peer_id, _err) in &failures {
-            self.notify_peer_failure(*peer_id);
+        // Notify consensus about failures if:
+        // 1. There is at least one success, otherwise it might be a problem of sending node
+        // 2. ???
```

Comment:
Maybe we should have a log below in the case where `successes.is_empty` warning that this peer cannot contact anyone for some reason.
This sounds like the node needs to be looked at.

Reply:
it will report an error to API